patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -292,18 +299,74 @@ public Recipe AddCondition(IEnumerable<Condition> conditions)
 		return this;
 	}
 
+	/// <summary>
+	/// Sets a condition delegate that will determine whether or not the recipe can be decrafted/changed in shimmer.. The condition can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="condition">The predicate delegate condition.</param>
+	/// <param name="description">A description of this condition. Use NetworkText.FromKey, or NetworkText.FromLiteral for this.</param>
+	public Recipe AddDecraftCondition(NetworkText description, Predicate<Recipe> condition) => AddDecraftCondition(new Condition(description, condition));
+
+	/// <summary>
+	/// Adds an array of conditions that will determine whether or not the recipe can be decrafted/changed in shimmer. The conditions can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="conditions">An array of conditions.</param>
+	public Recipe AddDecraftCondition(params Condition[] conditions) => AddDecraftCondition((IEnumerable<Condition>)conditions);
+
+	public Recipe AddDecraftCondition(Condition condition)
+	{
+		DecraftConditions.Add(condition);
+		return this;
+	}
+
+	/// <summary>
+	/// Adds a collection of conditions that will determine whether or not the recipe can be decrafted/changed in shimmer. The conditions can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="conditions">A collection of conditions.</param>
+	public Recipe AddDecraftCondition(IEnumerable<Condition> conditions)
+	{
+		DecraftConditions.AddRange(conditions);
+		return this;
+	}
+
+	/// <summary>
+	/// Adds every condition from Recipie.Condtions to Recipie.ShimmerContions, checking for duplicates.
+	/// </summary>
+	public Recipe CopyConditionsToShimmer()
+	{
+		foreach (Condition condition in Conditions) {
+			if (!DecraftConditions.Contains(condition)) {
+				DecraftConditions.Add(condition);
+			}
+		}
+		return this;
+	}
+
+	/// <summary>
+	/// Sets a check that is used during load to prevent shimmer being used for this recipe, not for on the fly use
+	/// </summary>
 	public Recipe DisableShimmer()
 	{
 		notDecraftable = true;
 		return this;
 	}
 
+	/// <summary>
+	/// resets a check that is used during load to prevent shimmer being used for this recipe, not for on the fly use
+	/// </summary>
+	public Recipe EnableShimmer()
+	{
+		notDecraftable = false;
+		return this;
+	}
+
+	[Obsolete("Replaced by the conditions system")]
```

Comment:
This and the other Obsolete attributes are unnecessary. The current iteration of shimmer code is 1.4-dev only. Simply remove them if they are unused.